lib/transmitter.js

```diff
@@ -1,21 +1,29 @@
 const debug = require('debug')('snyk:nodejs:transmitter');
 const needle = require('needle');
 
+const state = require('./state');
 const config = require('./config');
 const systemInfo = require('./system-info').getSystemInfo();
 
-const filters = {};
-const eventsToSend = [];
-
 function handlePeriodicTasks() {
   const url = config.beaconUrl;
   return transmitEvents(url, config.projectId, config.agentId);
 }
 
 function transmitEvents(url, projectId, agentId) {
+  const currentState = state.get();
   let postPromise = Promise.resolve();
-  debug(`agent:${agentId} transmitting ${eventsToSend.length} events to ${url} with project ID ${projectId}.`);
-  const body = {projectId, agentId, eventsToSend, systemInfo, filters};
+
+  debug(`agent:${agentId} transmitting ${currentState.events.length} events to ${url} with project ID ${projectId}.`);
+  const body = {
+    agentId,
+    projectId,
+    systemInfo,
+    filters: currentState.filters,
```

Comment:
I think you can just rename the keys in currentState to match the ones here and use `...currentState` instead of this, not critical

Reply:
I prefer to keep this abstraction:
the keys in "state" make sense for the Node agent while the keys in the transmission make sense for Homebase